## amd64-int3-fix.dpatch by <fschueler@gmx.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: Revert int3 handling
## DP: Patch author: Andi Kleen <ak@muc.de>
## DP: Upstream status: not yet committed to BK
#
--- a/arch/x86_64/kernel/kprobes.c	2005-06-06 11:22:29.000000000 -0400
+++ b/arch/x86_64/kernel/kprobes.c	2005-06-15 21:17:14.000000000 -0400
@@ -447,6 +447,8 @@
 	struct die_args *args = (struct die_args *)data;
 	switch (val) {
 	case DIE_INT3:
+		if (args->regs->cs & 3)
+			return NOTIFY_DONE;
 		if (kprobe_handler(args->regs))
 			return NOTIFY_STOP;
 		break;
